test/com/google/enterprise/adaptor/database/DatabaseAdaptorTest.java

```diff
@@ -561,4 +567,97 @@ public void testIncludeAllColumnsAsMetadataFalse_mcSet() throws Exception {
+
+    Map<String, String> configEntries = new HashMap<String, String>();
+    configEntries.put("db.user", "sa");
+    configEntries.put("db.password", "");
+    configEntries.put("db.url",
+        "jdbc:h2:mem:test;DB_CLOSE_DELAY=-1;DEFAULT_ESCAPE=");
+    configEntries.put("db.uniqueKey", "ID:int");
+    configEntries.put("db.everyDocIdSql", "select * from employees");
+    configEntries.put("db.singleDocContentSql",
+        "select * from employees where ID = ?");
+    configEntries.put("db.singleDocContentSqlParameters", "ID");
+    configEntries.put("db.aclSqlParameters", "ID");
+    configEntries.put("adaptor.namespace", "Default");
+    configEntries.put("db.modeOfOperation", "urlAndMetadataLister");
+    configEntries.put("db.modeOfOperation.urlAndMetadataLister.columnName",
+        "ID");
+    configEntries.put("docId.isUrl", "true");
+    configEntries.put("db.modeOfOperation.urlAndMetadataLister.displayUrlCol",
+        "ID");
+    configEntries.put("db.metadataColumns",
+        "ID:col1, FIRST_NAME:col2, LAST_NAME:col3");
+
+    final Config config = createStandardConfig(configEntries);
+    DatabaseAdaptor adaptor = new DatabaseAdaptor();
+    adaptor.init(TestHelper.createConfigAdaptorContext(config));
+
+    AccumulatingDocIdPusher pusher = new AccumulatingDocIdPusher();
+    adaptor.getDocIds(pusher);
+    List<Record> actual =  pusher.getRecords();
+
+    assertEquals(1, actual.size());
+
+    Metadata metadata = new Metadata();
+    metadata.add("col1",  "1001");
+    metadata.add("col2",  "John");
+    metadata.add("col3",  "Doe");
+    ImmutableList.Builder<Record> builder = ImmutableList.builder();
+    builder.add(new Record.Builder(new DocId("1001")).setMetadata(metadata)
+        .build());
+    List<Record> expected = builder.build();
+
+    assertEquals(expected, actual);
+  }
+
+  @Test
+  public void testGetDocContent() throws Exception {
+    addDBTableForTest();
+
+    Map<String, String> configEntries = new HashMap<String, String>();
+    configEntries.put("db.user", "sa");
+    configEntries.put("db.password", "");
+    configEntries.put("db.url",
+        "jdbc:h2:mem:test;DB_CLOSE_DELAY=-1;DEFAULT_ESCAPE=");
+    configEntries.put("db.everyDocIdSql", "select * from employees");
+    configEntries.put("db.singleDocContentSql",
+        "select * from employees where ID = ?");
+    configEntries.put("adaptor.namespace", "Default");
+    configEntries.put("db.modeOfOperation", "rowToText");
+    configEntries.put("db.metadataColumns",
+        "ID:col1, FIRST_NAME:col2, LAST_NAME:col3");
+
+    final Config config = createStandardConfig(configEntries);
+    DatabaseAdaptor adaptor = new DatabaseAdaptor();
+    adaptor.init(TestHelper.createConfigAdaptorContext(config));
+
+    MockRequest request = new MockRequest(new DocId("1001"));
+    MockResponse response = new MockResponse();
+    adaptor.getDocContent(request, response);
+
+    assertEquals("col1", adaptor.metadataColumns.getMetadataName("ID"));
```

Comment:
Why a different set of assertions here compared to the other test? I like the other one better.